### PR TITLE
Add examples/nomad folder

### PR DESCRIPTION
**Pull request by avillela**
**Description:**
For those wishing to run the OTel Collector on HashiCorp Nomad. Add examples/nomad folder with instructions on how to run the OTel Collector on Nomad using Traefik as the load balancer.

Added `otel-collector.nomad` jobspec and `traefik.nomad` jobspec.

**Link to tracking Issue:** n/a

**Testing:** Tested as per following [this tutorial](https://storiesfromtheherd.com/just-in-time-nomad-running-the-opentelemetry-collector-on-hashicorp-nomad-with-hashiqube-4eaf009b8382).

**Documentation:** Added readme linking to comprehensive tutorial explaining how to run this locally in Nomad.

### CODE REVIEW

**Comment by dmitryax** [comment]
@avillela thanks for the contribution. Please sign the CLA

**Comment by avillela** [comment]
> @avillela thanks for the contribution. Please sign the CLA

I tried like 3x :(

File: examples/nomad/otel-collector.nomad [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+      - $DATADOG_TAG_NAME
+    api:
+      key: "{{ with secret "kv/data/otel/o11y/datadog" }}{{ .Data.data.api_key }}{{ end }}"
+      site: datadoghq.com
+
+service:
+  pipelines:
+    traces:
+      receivers: [otlp]
+      exporters: [logging, otlp/ls, otlp/hc, datadog]
```
Comment (dmitryax) [inline examples/nomad/otel-collector.nomad:134]: It's unlikely that users want to send data to 3 tracing backend providers. So some changes has to be done to this example to play with this it. Can we remove vendor exporters and keep `logging` only?

Comment (avillela) [reply examples/nomad/otel-collector.nomad]: @dmitryax Although my goal with this was to show how to configure 3 different Observability back-ends, I see your point. I've updated the file as per above.

**Review by dmitryax — APPROVED** [review]

**Review by codeboten — APPROVED** [review]

**Pull request merged**